core/services/relay/evm/event_binding.go

```diff
@@ -121,7 +121,7 @@ func (e *eventBinding) QueryKey(ctx context.Context, filter query.KeyFilter, lim
 	}
 	remapped.Expressions = append(defaultExpressions, remapped.Expressions...)
 
-	logs, err := e.lp.FilteredLogs(ctx, remapped, limitAndSort)
+	logs, err := e.lp.FilteredLogs(ctx, remapped, limitAndSort, e.contractName+"-"+e.eventName)
```

Comment:
I am not sure if this is enough from observability perspective. We can query for same event from same contract with different filters. We should either pass a queryName from `QueryKey` or somehow turn limitAndSort and filter into a string? Although to make it easier we can do this in a separate PR under a separate ticket.

@mateusz-sekara thought?

Reply:
I think a separate PR would be best to iron out the rest of this detail.